tests/mujoco/test_util.py

```diff
@@ -59,20 +58,20 @@ def _transfer_state(env_name):
     env2.reset()
     handler.set_env_state(state, env2)
 
-
 def test_freeze():
-    _freeze_mujoco_gym_env("gym___HalfCheetah-v2")
-    _freeze_mujoco_gym_env("gym___Hopper-v2")
-    _freeze_mujoco_gym_env("gym___Humanoid-v2")
+    # TODO(Rohan138): This is flaky for some reason?
```

Comment:
FYI still TODO here.

Reply:
This is a *very* weird bug: HalfCheetah is the only one that's flaky, and fails about 1/3 times I test it. I'm getting mildly suspicious of the underlying gymnasium env at this point, might open an issue there in a bit.

Reply:
Update: Testing with `for run in {1..10}; do pytest -rP tests/mujoco/test_util.py -k "test_freeze"; done`, of the 10 gym environments, only `Ant-v4`, `HalfCheetah-v4`, and `HumanoidStandup-v4` are failing.

Reply:
What error are you getting?

Reply:
```
env_name = 'gym___Ant-v4'

    def _freeze_mujoco_gym_env(env_name):
        handler = create_handler_from_str(env_name)
        env = handler.make_env_from_str(env_name)
        env.reset(seed=0)
    
        seen_obses = []
        seen_rewards = []
        actions = []
        num_steps = 100
    
        with handler.freeze(env):
            for _ in range(num_steps):
                action = env.action_space.sample()
                next_obs, reward, terminated, truncated, _ = env.step(action)
                seen_obses.append(next_obs)
                seen_rewards.append(reward)
                actions.append(action)
                if terminated or truncated:
                    break
    
        for a in actions:
            next_obs, reward, _, _, _ = env.step(a)
            ref_obs = seen_obses.pop(0)
            ref_reward = seen_rewards.pop(0)
            np.testing.assert_array_almost_equal(next_obs, ref_obs)
>           assert reward == pytest.approx(ref_reward)
E           assert 0.14005772492059787 == 0.14005799237443473 ± 1.4e-07
E             comparison failed
E             Obtained: 0.14005772492059787
E             Expected: 0.14005799237443473 ± 1.4e-07

tests/mujoco/test_util.py:37: AssertionError
```

Reply:
The values are always slightly off, but I've tried decreasing the tolerance on the comparison and it only becomes less flaky, not 100% passing. Plus I really shouldn't need to do that logically, based on what the `freeze` is doing.

Reply:
I made a simple reproducible example of this issue and it seems to be isolated to gymnasium. Opened an issue: https://github.com/Farama-Foundation/Gymnasium/issues/352

Reply:
Given that this bug is gymnasium/mujoco specific, can we keep the flakey envs commented out and open a separate issue?

Reply:
Does this issue break the CEM-based controller (on gt dynamics)? Or is it just affecting the test?

Reply:
I don't think the freeze wrapper is used there (or anywhere, actually), so the controller should be fine.

Reply:
It's used by the controller for ground truth dynamics, under `diagnostics/control_env.py`. 

In [this](https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/diagnostics/control_env.py#L59) line we call `EnvHandler.rollout_env`, which in turn calls each class' `freeze` method [here](https://github.com/facebookresearch/mbrl-lib/blob/3c6553a43e8fea37b0695dacab91e3fefb147ad7/mbrl/util/env.py#L261). And for Mujoco this is defined [here](https://github.com/facebookresearch/mbrl-lib/blob/3c6553a43e8fea37b0695dacab91e3fefb147ad7/mbrl/util/mujoco.py#L71), which I believe is the same freeze called in the unit test.

Reply:
> Does this issue break the CEM-based controller (on gt dynamics)? Or is it just affecting the test?
@luisenp 

From the tests, it seem that it affects the specific envs `Ant-v4, HalfCheetah-v4',HumanoidStandup-v4` as it doesn't affect the `Hopper-v4,Humanoid-v4`. So I can plan to run the `control_env.py` on all those envs and compare the results to `main` and see what happens. 

If it breaks on the known affected envs, I can open a separate issue after we merge this PR detailing that those envs do not work correctly and link the gymnasium issue and wait for that fix to resolve this issue.

Reply:
I agree with that plan. Thanks @raghavauppuluri13!

Reply:
@luisenp @Rohan138  These are the results. One thing to note is that we're comparing the v2 versions with the v4 versions between main and diff respectively. This concludes that all are training, but the performance is slightly different between the versions. Does that look fine? Still worth making an issue to track the gymnasium issue that was made with these curves if people find slight issues with reproduction.
![image](https://user-images.githubusercontent.com/41026849/223532874-5575e75c-209c-437e-8fb8-d02c56165b8b.png)

Reply:
Yeah, this is good. Thanks a lot @raghavauppuluri13! I want to give a final pass before merging, but I don't expect much changes to be necessary. I'll try to get this in by Thursday. Thanks!